pkg/router/controller/host_admitter_test.go

```diff
@@ -0,0 +1,399 @@
+package controller
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/watch"
+
+	routeapi "github.com/openshift/origin/pkg/route/api"
+	"github.com/openshift/origin/pkg/route/api/validation"
+)
+
+type rejectionRecorder struct {
+	rejections map[string]string
+}
+
+func (_ rejectionRecorder) rejectionKey(route *routeapi.Route) string {
+	return route.Namespace + "-" + route.Name
+}
+
+func (r rejectionRecorder) RecordRouteRejection(route *routeapi.Route, reason, message string) {
+	r.rejections[r.rejectionKey(route)] = reason
+}
+
+func admissionChecker(denyDomain string, allowWildcards bool) RouteAdmissionFunc {
```

Comment:
Given that the only point of the admitter function in this file's testing is to ensure that HandleRoute executes route rejection, consider using an anonymous function that statically returns ``nil`` or ``err``.